src/mongocxx/database.hpp

```diff
@@ -248,6 +248,7 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const std::unique_ptr<impl>& get_impl() const;
```

Comment:
Then should get_impl() be const? This doesn't seem right, since it gives callers a way to alter the impl's state. However, if it's not const, then it can't be used in const functions.